tools/dashboard-plugin-template/src/harness/PluginLoader.tsx

```diff
@@ -1,23 +1,64 @@
 // (C) 2019-2022 GoodData Corporation
-import React from "react";
+import React, { useCallback, useMemo } from "react";
 import { idRef } from "@gooddata/sdk-model";
-import { DashboardLoadStatus, IEmbeddedPlugin, useDashboardLoaderWithReload } from "@gooddata/sdk-ui-loaders";
+import { IEmbeddedPlugin, useDashboardLoaderWithPluginManipulation } from "@gooddata/sdk-ui-loaders";
 import PluginFactory from "../plugin";
 import { DEFAULT_DASHBOARD_ID } from "./constants";
-import { DashboardConfig } from "@gooddata/sdk-ui-dashboard";
+import {
+    DashboardConfig,
+    CustomToolbarComponent,
+    DefaultDashboardToolbar,
+    DefaultDashboardToolbarButton,
+    DefaultDashboardToolbarGroup,
+} from "@gooddata/sdk-ui-dashboard";
 import { ErrorComponent, LoadingComponent } from "@gooddata/sdk-ui";
-import { Toolbar } from "./Toolbar";
 
 const Plugins: IEmbeddedPlugin[] = [{ factory: PluginFactory }];
 const Config: DashboardConfig = { mapboxToken: process.env.MAPBOX_TOKEN };
 const DashboardRef = idRef(process.env.DASHBOARD_ID || DEFAULT_DASHBOARD_ID, "analyticalDashboard");
 
-interface IPluginLoaderBodyProps {
-    loaderStatus: DashboardLoadStatus;
-}
+export const PluginLoader = () => {
+    const { loaderStatus, reloadPlugins, setExtraPlugins, extraPlugins } =
+        useDashboardLoaderWithPluginManipulation({
+            dashboard: DashboardRef,
+            loadingMode: "staticOnly",
+            config: Config,
+            extraPlugins: Plugins,
+        });
+
+    const isPluginEnabled = !!extraPlugins?.length;
+
+    const togglePlugin = useCallback(() => {
+        if (isPluginEnabled) {
+            setExtraPlugins([]);
+        } else {
+            setExtraPlugins(Plugins);
+        }
+    }, [isPluginEnabled, setExtraPlugins]);
+
+    const { status, error, result } = loaderStatus;
 
-const PluginLoaderBody: React.FC<IPluginLoaderBodyProps> = (props) => {
-    const { status, error, result } = props.loaderStatus;
+    const ToolbarComponent = useMemo<CustomToolbarComponent>(() => {
+        return function CustomToolbar() {
```

Comment:
It will separate it from memo into component.

Reply:
We cannot completely avoid this, but I separated the body in a separate component.